File: src/Python/task_1/pattern.py
# Функция для печати алфавитной пирамиды в обратном порядке
def print_alphabet_pyramid(N):
    # Итерация от n до 1 для печати строк сверху вниз
    for row in range(N, 0, -1):
        # Итерация от n до row для печати символов в строке справа налево
        for letter in range(N, row - 1, -1):
            # Печать символа, начиная с соответствующего номеру строки
            print(chr(ord('A') + letter - 1), end=' ')
        # Переход на новую строку
        print()

N = int(input())
print_alphabet_pyramid(N)

File: src/Haskell/task_1/pattern.hs
import Data.Char (chr, ord)

-- Функция для печати алфавитного пирамиды
printAlphabetPyramid :: Int -> IO ()
printAlphabetPyramid n = mapM_ printRow [n, n-1 .. 1]
  where
    printRow row = do
      putStrLn $ concatMap (\letter -> [chr (ord 'A' + letter - 1), ' ']) [n, n-1 .. row]

-- Главная функция для ввода
main :: IO ()
main = do
  input <- getLine
  let n = read input :: Int
  printAlphabetPyramid n